hls4ml/model/hls_layers.py

```diff
@@ -1513,14 +1513,18 @@ def initialize(self):
         self.set_attr('dim', '{}d'.format(len(inp.shape)))
         if len(perm) == 4 and perm[0] == 0:
             perm = [i - 1 for i in perm[1:]]
+        elif len(perm) == 3 and len(inp.shape) == 2 and perm[0] == 0:
```

Comment:
This check shouldn't be needed. Even the previous one for 3d looks like a bad idea. We expect `perm` to be zero-based, not to do extra work after conversion.

Reply:
Yes, I think this logic can be removed